### PR TITLE
Uses the dialog title theme

### PR DESCRIPTION
**Pull request by ademar111190**
I'm using the title style on the dialog theme to address https://github.com/breez/c-breez/issues/412

Breez's `PaymentInfo` model has two fields, `dialog title` and `description`, that we present using two different styles, title and description. C-breez `Payment` model has a description field, but it doesn't have anything else on itself or its attached fields.

I would like to know if you have another solution in mind @roeierez @erdemyerebasmaz; so far, what I have done here is to remove the description style and uses the title style, despite the title content coming from the description field.

### CODE REVIEW

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart [hunk 1]
```diff
@@ -7,12 +7,12 @@ import 'package:c_breez/models/currency.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
-class PaymentDetailsAmount extends StatelessWidget {
```
Comment (ademar111190) [inline lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart:10]: Just a minor fix on the name to fit breezmobile component and its file name

File: lib/routes/home/widgets/payments_list/payment_details_dialog.dart [hunk 2]
```diff
@@ -36,10 +36,10 @@ Future<void> showPaymentDetailsDialog(
             mainAxisSize: MainAxisSize.min,
             crossAxisAlignment: CrossAxisAlignment.stretch,
             children: [
-              PaymentDetailsDialogDescription(
```
Comment (erdemyerebasmaz) [inline lib/routes/home/widgets/payments_list/payment_details_dialog.dart:39]: I don't think the description widget should be removed, that was not what I had in mind. I was suggesting we use the extractTitle method you introduced on title widget and display the full description on description widget.

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart (hunk 1 above)
Comment (erdemyerebasmaz) [reply lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart:10]: I suggest we use shorter and singular names for items we display on PaymentDetailsDialog in the form of: PaymentDetail{Parameter}.
_PaymentDetailTitle, PaymentDetailPreimage, PaymentDetailAmount etc._

Comment (ademar111190) [reply lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart:10]: I want to keep the same name on BreezMobile. There are more widgets, and the more verbose name has a reason to differentiate from other widgets

Comment (erdemyerebasmaz) [reply lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart:10]: Last time I checked these were not extracted as standalone widgets on Breezmobile but I see what you mean, it's just a bit tiring on the eyes that's all :) They can stay as they are.

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_description.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    super.key,
+    required this.paymentInfo,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final texts = context.texts();
+    final themeData = Theme.of(context);
+
+    final description = paymentInfo.extractTitle(texts);
```
Comment (erdemyerebasmaz) [inline lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_description.dart:20]: I was suggesting to use description as it is here.

File: lib/routes/home/widgets/payments_list/payment_details_dialog.dart (hunk 2 above)
Comment (ademar111190) [reply lib/routes/home/widgets/payments_list/payment_details_dialog.dart:39]: I just added the subtitle back; it is showing the same text as the title as it is coming from the same property, the only differences are the style and the ellipses/multiline strategy to long texts.

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_description.dart (hunk 3 above)
Comment (ademar111190) [reply lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_description.dart:20]: Yep, I did it.

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart (hunk 1 above)
Comment (ademar111190) [reply lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_amount.dart:10]: You are right; the PR is still open https://github.com/breez/breezmobile/pull/1048

If you want to suggest names there, I can apply the same names here later; the only thing is there are more widgets so the naming can be thought with better context.

**Review by erdemyerebasmaz — APPROVED** [review]

**Pull request merged**